spec/check_spec.lua

```diff
@@ -779,10 +779,31 @@ return foo;
       }, check[[
 local foo, bar -- luacheck: ignore foo
 for _ in pairs({}) do return end -- luacheck: ignore
-      ]])
+]])
    end)
 
    it("handles argparse sample", function()
       assert.table(check(io.open("spec/samples/argparse.lua", "rb"):read("*a")))
    end)
+
+   it("detects whitespace issues", function()
+      local warnings = check(
+         io.open("spec/samples/bad_whitespace.lua", "rb"):read("*a")
```

Comment:
Tests in check_spec mostly operate on strings. Checks on files go into cli_spec, where error messages are also tested.